File: Assets/scripts/fixOnClick.js
#pragma strict

var rate : float = 15; // rate healed per second
var repairMazeGenerator : GameObject;

function Start() {
	repairMazeGenerator = GameObject.FindGameObjectWithTag("RepairMazeGenerator");
}


function Update () {
	var hit : RaycastHit; 
	if (Input.GetMouseButtonUp(0)){
		if(Physics.Raycast(Camera.main.ScreenPointToRay(Input.mousePosition), hit)) {
			var damageComponent : damageableComponent = hit.transform.gameObject.GetComponent("damageableComponent");
			if (damageComponent) {
				repairMazeGenerator.SendMessage("StartMaze", hit.transform.gameObject);
//				damageComponent.Repair(rate * Time.deltaTime);
//				Debug.Log("Repairing: " + hit.transform.gameObject);
			}
		}
	}
}

File: Assets/scripts/c src/FixOnClick.cs
using UnityEngine;
using System.Collections;

public class FixOnClick : MonoBehaviour {

	public GameObject repairMazeGenerator;

	// Use this for initialization
	void Start () {
		repairMazeGenerator = GameObject.FindGameObjectWithTag ("RepairMazeGenerator");
	}
	
	// Update is called once per frame
	void Update () {
		RaycastHit hit; 
		if (Input.GetMouseButtonUp(0)){
			if(Physics.Raycast(Camera.main.ScreenPointToRay(Input.mousePosition), out hit)) {
				DamageableComponent damageComponent = hit.transform.gameObject.GetComponent<DamageableComponent>();
				if (damageComponent) {
					repairMazeGenerator.SendMessage("StartMaze", hit.transform.gameObject);
					//				damageComponent.Repair(rate * Time.deltaTime);
					//				Debug.Log("Repairing: " + hit.transform.gameObject);
				}
			}
		}
	}
}
